tests/dryrun/install_test.go

```diff
@@ -586,3 +586,171 @@ func TestNoDomains(t *testing.T) {
 
 	t.Logf("%s: test complete", time.Now().Format(time.RFC3339))
 }
+
+// this test is to ensure that http proxy settings are passed through correctly
+func TestInstall_HTTPProxy(t *testing.T) {
+	t.Setenv("HTTP_PROXY", "http://localhost:3128")
+	t.Setenv("HTTPS_PROXY", "http://localhost:3128")
+	t.Setenv("NO_PROXY", "localhost,127.0.0.1,10.0.0.0/8")
+
+	hostCABundle := findHostCABundle(t)
+
+	hcli := &helm.MockClient{}
+
+	mock.InOrder(
+		// 4 addons
+		hcli.On("Install", mock.Anything, mock.Anything).Times(4).Return(nil, nil),
+		hcli.On("Close").Once().Return(nil),
+	)
+
+	dr := dryrunInstall(t, &dryrun.Client{HelmClient: hcli})
+
+	// --- validate addons --- //
+
+	// embedded cluster operator
+	assert.Equal(t, "Install", hcli.Calls[1].Method)
+	operatorOpts := hcli.Calls[1].Arguments[1].(helm.InstallOptions)
+	assert.Equal(t, "embedded-cluster-operator", operatorOpts.ReleaseName)
+
+	// NO_PROXY is calculated
+	val, err := helm.GetValue(operatorOpts.Values, "extraEnv")
+	require.NoError(t, err)
+	var noProxy string
+	for _, v := range val.([]map[string]any) {
+		if v["name"] == "NO_PROXY" {
+			noProxy = v["value"].(string)
+		}
+	}
+	assert.NotEmpty(t, noProxy)
+	assert.Contains(t, noProxy, "10.0.0.0/8")
+
+	assertHelmValues(t, operatorOpts.Values, map[string]any{
+		"extraEnv": []map[string]any{
+			{
+				"name":  "HTTP_PROXY",
+				"value": "http://localhost:3128",
+			},
+			{
+				"name":  "HTTPS_PROXY",
+				"value": "http://localhost:3128",
+			},
+			{
+				"name":  "NO_PROXY",
+				"value": noProxy,
+			},
+		},
+	})
+	// TODO: CA
+
+	// velero
+	assert.Equal(t, "Install", hcli.Calls[2].Method)
+	veleroOpts := hcli.Calls[2].Arguments[1].(helm.InstallOptions)
+	assert.Equal(t, "velero", veleroOpts.ReleaseName)
+	assertHelmValues(t, veleroOpts.Values, map[string]any{
+		"configuration.extraEnvVars": map[string]any{
+			"HTTPS_PROXY":  "http://localhost:3128",
+			"HTTP_PROXY":   "http://localhost:3128",
+			"NO_PROXY":     noProxy,
+			"SSL_CERT_DIR": "/certs",
+		},
+		"extraVolumes": []map[string]any{{
+			"name": "host-ca-bundle",
+			"hostPath": map[string]any{
+				"path": hostCABundle,
+				"type": "FileOrCreate",
+			},
+		}},
+		"extraVolumeMounts": []map[string]any{{
+			"mountPath": "/certs/ca-certificates.crt",
+			"name":      "host-ca-bundle",
+		}},
+	})
+
+	// admin console
+	assert.Equal(t, "Install", hcli.Calls[3].Method)
+	adminConsoleOpts := hcli.Calls[3].Arguments[1].(helm.InstallOptions)
+	assert.Equal(t, "admin-console", adminConsoleOpts.ReleaseName)
+	assertHelmValues(t, adminConsoleOpts.Values, map[string]any{
+		"extraEnv": []map[string]any{
+			{
+				"name":  "ENABLE_IMPROVED_DR",
+				"value": "true",
+			},
+			{
+				"name":  "HTTP_PROXY",
+				"value": "http://localhost:3128",
+			},
+			{
+				"name":  "HTTPS_PROXY",
+				"value": "http://localhost:3128",
+			},
+			{
+				"name":  "NO_PROXY",
+				"value": noProxy,
+			},
+		},
+	})
+	// TODO: CA
```

Comment:
let's add these as tasks in the "mount ca bundle into kotsadm" ticket?

Reply:
added